tests/affiliatedcertification/affiliatedcerthelper/affiliatedcerthelper.go

```diff
@@ -0,0 +1,58 @@
+package affiliatedcerthelper
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	"github.com/test-network-function/cnfcert-tests-verification/tests/affiliatedcertification/affiliatedcertparameters"
+	"github.com/test-network-function/cnfcert-tests-verification/tests/globalhelper"
+	"github.com/test-network-function/cnfcert-tests-verification/tests/globalparameters"
+	"github.com/test-network-function/cnfcert-tests-verification/tests/networking/nethelper"
+	"github.com/test-network-function/cnfcert-tests-verification/tests/networking/netparameters"
+)
+
+func SetUpAndRunContainerCertTest(containersInfo []string, expectedResult string) error {
+	var err error
+
+	ginkgo.By("Add container information to " + globalparameters.DefaultTnfConfigFileName)
+
+	err = globalhelper.DefineTnfConfig(
+		[]string{netparameters.TestNetworkingNameSpace},
+		[]string{netparameters.TestPodLabel},
+		containersInfo)
+
+	if err != nil {
+		return fmt.Errorf("error defining tnf config file: %w", err)
+	}
+
+	ginkgo.By("Start test")
+
+	err = globalhelper.LaunchTests(
+		[]string{affiliatedcertparameters.AffiliatedCertificationTestSuiteName},
+		affiliatedcertparameters.TestCaseContainerSkipRegEx,
+	)
+
+	if strings.Contains(expectedResult, globalparameters.TestCaseFailed) && err == nil {
+		return fmt.Errorf("error running %s test",
+			affiliatedcertparameters.AffiliatedCertificationTestSuiteName)
+	}
+
+	if (strings.Contains(expectedResult, globalparameters.TestCasePassed) ||
```

Comment:
No sure I understand the reason for this test and the one above. Shouldn't err simply be always nil, irrespective of the expected result? if the exec cmd fails, success and failure results should be equally not valid. Maybe I am missing something else?

Reply:
The first case (TestCaseFailed and error is nil) if for negative tests: our expected outcome is an error, so the case where an error isn't present is the one we need to throw an error about in the testing.

Reply:
I see, thanks. The exit code is actually non zero in case test cases fail